xlml/utils/subprocess_utils.py

```
@@ -82,3 +83,23 @@ def run_exec(
+
+
+def run_streaming(
+    cmd: str, log_command: bool = True, env: dict[str, str] | None = None
+) -> None:
+  """Executes a shell command and streams its output in real-time."""
+
+  if log_command:
+    logging.info("[subprocess] executing command:\n %s\n", cmd)
+
+  # Use subprocess.Popen to stream output
+  return subprocess.Popen(
```

Ruff (0.16.1)

[error] `subprocess` call with `shell=True` identified, security issue

(S602)